File: InsuranceApi/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace InsuranceApi.Models;
public class Vehicle
{
    public Guid Id { get; set; }

    [Required]
    [MaxLength(50)]
    public string RegistrationNumber { get; set; } = string.Empty;

    [MaxLength(100)]
    public string Make { get; set; } = string.Empty;

    [MaxLength(100)]
    public string Model { get; set; } = string.Empty;

    public int Year { get; set; }

    public VehicleType Type { get; set; } // Private or Commercial
}

public enum VehicleType
{
    Private,
    Commercial
}

File: motor-insurance-frontend/src/services/api/vehicle.ts
import api from "./axios";

export interface Vehicle {
  id?: string;
  registrationNumber: string;
  make: string;
  model: string;
  colour: string;
  year: number;
  type: number; // 0 = Private, 1 = Commercial
}

export async function searchVehicle(registrationNumber: string) {
  return api.get(`/Vehicles/search/${registrationNumber}`);
}

export async function addVehicle(vehicle: Vehicle) {
  return api.post("/Vehicles", vehicle);
}
